pkg/credhelper/helper.go

```diff
@@ -18,14 +18,21 @@ package credhelper
 import (
 	"errors"
 	"fmt"
+	"net/url"
+	"regexp"
 
 	"github.com/Azure/go-autorest/autorest/azure/auth"
 	"github.com/chrismellard/docker-credential-acr-env/pkg/registry"
 	"github.com/chrismellard/docker-credential-acr-env/pkg/token"
 	"github.com/docker/docker-credential-helpers/credentials"
 )
 
-const tokenUsername = "<token>"
+var acrPattern = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.azurecr.io$`)
```

Comment:
See the k8s regex

https://github.com/vdemeester/k8s-pkg-credentialprovider/blob/802b5fd6c9a9c043749089680a492c8fac9e38a3/azure/azure_credentials.go#L53-L54

There are other 'regions'

Reply:
Very helpful! Thanks!